### PR TITLE
Some IO/mixer updates

### PR DESCRIPTION
**Pull request by AndreasAntener**
- fixes setting/getting trims with heli and null mixers
- fixes sbus output rate update (was called when setting other registers)
- (nice to have) only sets trims when changed, re-use global check function in heli mixer

See single commits

### CODE REVIEW

**Comment by dagar** [comment]
I don't know the px4iofirmware side very well, but overall this looks good.

Can you comment on testing?

**Comment by AndreasAntener** [comment]
Bench tested on master
- setting trims and verified that they get set while running and after a reboot
- checked that the heli mixer loads

(flown on our branch)

I couldn't test the sbus rate setting, but that seemed to be obviously wrong before.

**Comment by AndreasAntener** [comment]
Of course tested with mixers that include null channels, as that was the problem I stumbled upon.

File: src/lib/mixer/mixer_helicopter.cpp [hunk 1]
```diff
@@ -65,22 +65,9 @@ HelicopterMixer::from_text(Mixer::ControlCallback control_cb, uintptr_t cb_handl
...
-		if (buf[i] == ' ' || buf[i] == '\n' || buf[i] == '\r') {
-			/* found a line ending or space, so no split symbols / numbers. good. */
-			break;
-
-		} else {
-			debug("simple parser rejected: No newline / space at end of buf. (#%d/%d: 0x%02x)", i, buflen - 1, buf[i]);
-			return nullptr;
-		}
-
+	/* enforce that the mixer ends with a new line */
```
Comment (LorenzMeier) [inline src/lib/mixer/mixer_helicopter.cpp]: What is the intention of this change?

File: src/modules/px4iofirmware/registers.c [hunk 2]
```diff
@@ -688,33 +692,27 @@ registers_set_one(uint8_t page, uint8_t offset, uint16_t value)
...
+		case PX4IO_P_SETUP_TRIM_PITCH:
+		case PX4IO_P_SETUP_TRIM_YAW:
+		case PX4IO_P_SETUP_SCALE_ROLL:
+		case PX4IO_P_SETUP_SCALE_PITCH:
+		case PX4IO_P_SETUP_SCALE_YAW:
+		case PX4IO_P_SETUP_MOTOR_SLEW_MAX:
+		case PX4IO_P_SETUP_AIRMODE:
 		case PX4IO_P_SETUP_THERMAL:
-			r_page_setup[PX4IO_P_SETUP_THERMAL] = value;
+			r_page_setup[offset] = value;
```
Comment (LorenzMeier) [inline src/modules/px4iofirmware/registers.c]: Nice cleanup!

File: src/lib/mixer/mixer_helicopter.cpp (hunk 1 above)
Comment (AndreasAntener) [reply src/lib/mixer/mixer_helicopter.cpp]: Only cleanup. A while ago this chunk of code got removed in the other mixers and replaced with the library function.

**Comment by LorenzMeier** [comment]
Thanks!

**Pull request merged**